crates/partition-store/src/partition_store_manager.rs

```diff
@@ -194,6 +198,52 @@ impl PartitionStoreManager {
         Ok(partition_store)
     }
 
+    pub async fn export_partition_snapshot(
+        &self,
+        partition_id: PartitionId,
+        snapshot_id: SnapshotId,
+        snapshot_base_path: PathBuf,
+    ) -> anyhow::Result<LocalPartitionSnapshot> {
+        let mut guard = self.lookup.lock().await;
+
+        let partition_store = guard
+            .live
+            .get_mut(&partition_id)
+            .ok_or(anyhow!("Unknown partition: {}", partition_id))?;
+
+        // RocksDB will create the snapshot directory but the parent must exist first:
+        tokio::fs::create_dir_all(&snapshot_base_path).await?;
+        let snapshot_dir = snapshot_base_path.join(snapshot_id.to_string());
+
+        Ok(partition_store.create_snapshot(snapshot_dir).await?)
+    }
+
+    pub async fn update_archived_lsn(
+        &self,
+        partition_id: PartitionId,
+        archived_lsn: Lsn,
+    ) -> anyhow::Result<()> {
+        let mut guard = self.lookup.lock().await;
+
+        let partition_store = guard
+            .live
+            .get_mut(&partition_id)
+            .ok_or(anyhow!("No store for partition: {}", partition_id))?;
+
+        let previous_archived_snapshot_lsn = partition_store.get_archived_lsn().await?;
+        let mut tx = partition_store.transaction();
+        tx.put_archived_lsn(archived_lsn).await;
+        tx.commit().await?;
```

Comment:
Since these are forced to be non-WAL writes in the writer, we might actually lose this update if there is no flush! (I've noticed that ctrl-c loses the unflushed data, even on otherwise clean shutdown.)